src/dodal/beamlines/i18.py

```diff
@@ -0,0 +1,263 @@
+) -> DoubleCrystalMonochromator:
+    return device_instantiation(
+        DoubleCrystalMonochromator,
+        "dcm",
+        "",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        bl_prefix=False,
+        motion_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-MO-DCM-01:",
+        temperature_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-DI-DCM-01:",
+        crystal_1_metadata=crystal_1_metadata,
+        crystal_2_metadata=crystal_2_metadata,
+    )
+
+
+@skip_device()
+def i0(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> TetrammDetector:
+    return device_instantiation(
+        TetrammDetector,
+        "i0",
+        "-DI-XBPM-02:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        type="Cividec Diamond XBPM",
+        path_provider=get_path_provider(),
+    )
+
+
+@skip_device()
+def it(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> TetrammDetector:
+    return device_instantiation(
+        TetrammDetector,
+        "it",
+        "-DI-XBPM-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        type="Tetramm",
+        path_provider=get_path_provider(),
+    )
+
+
+@skip_device
+def vfm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> KBMirror:
+    return device_instantiation(
+        KBMirror,
+        "vfm",
+        "-OP-VFM-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+def hfm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> KBMirror:
+    return device_instantiation(
+        KBMirror,
+        "hfm",
+        "-OP-HFM-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+def diode(wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False) -> Diode:
+    return device_instantiation(
+        Diode,
+        "diodad7bdiode",
+        "-DI-PHDGN-07:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+def table(wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False) -> Table:
+    return device_instantiation(
+        Table,
+        "table",
+        "-MO-TABLE-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+# note this is a mock table, not sure how does it relate to the real Table,
+# maybe just a fake option in instantiation is needed
+@skip_device()
+def raster_stage(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> RasterStage:
+    return device_instantiation(
+        RasterStage,
+        "raster_stage",
+        "-MO-SIM-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+def sim_detector(wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False):
+    return device_instantiation(
+        SimDetector,
+        "sim_detector",
+        "-MO-SIM-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
```

Comment:
```suggestion
```

Not a device on i18.

Reply:
we'd like one for testing - as discussed with @iain-hall